src/librustc_typeck/check/method/suggest.rs

```diff
@@ -282,6 +283,18 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                                               item_name,
                                               rcvr_expr,
                                               out_of_scope_traits);
+
+                if !lev_candidates.is_empty() {
+                    use rustc_data_structures::fx::FxHashSet;
+                    let mut dedup = FxHashSet();
+                    let limit = if lev_candidates.len() == 5 { 5 } else { 4 };
```

Comment:
We don't do the "and K others" thing here, so there's no point in having the dynamic limit.

Reply:
Ok, removed.

Reply:
The idea was to still limit it to 5 candidates, just not "either 5 or 4".

Reply:
Oh, my bad, I'll fix it. I thought the idea of the limit was just to avoid leaving out one other candidate, so display by default up to four, unless there's five in which case we'll display the fifth as well.

Reply:
Actually, in most places in the compiler we find the 1 best name that matches, using `find_best_match_for_name` from the list of names of candidates.